content/en/blog/2023/ambient-performance-savings/index.md

```diff
@@ -0,0 +1,135 @@
+
+In the ambient scenarios, each ztunnel instance sees small spikes as they handle cross-node traffic for the different requests. These ztunnel spikes depend on which nodes the Fortio client and httpbin server reside on as the different versions are hit. In the ambient with both ztunnel and waypoint proxy scenario, there are clear spikes in the waypoint proxies as a particular version of httpbin is called since one waypoint captures traffic for all instances of that version.
+
+Though the waypoints consume similar resources as sidecars, the Rust-based ztunnel has a much smaller CPU utilization. An individual ztunnel use *less than 20%* in comparison to a sidecar and all three ztunnels for the entire cluster combined use less than a single sidecar!
+
+Next is memory usage by pod. In all Istio scenarios, memory usage stays relatively constant for each pod during the test runs. In both ambient scenarios, the L4 ztunnel consumes so little memory it’s almost hard to display them on the same scale as sidecar usage. Waypoint proxies consume a similar amount of resources as sidecars do, with only minor improvements. So what do these per pod usage improvements mean for the totals across the cluster? Everything.
+
+{{< image width="100%"
+    link="total-cpu.png"
+    caption="Total CPU usage for sidecar and ambient scenarios"
+    >}}
+
+{{< image width="100%"
+    link="total-memory.png"
+    caption="Total memory usage for sidecar and ambient scenarios"
+    >}}
+
+{{< image width="100%"
+    link="cpu-by-pod.png"
+    caption="Stacked CPU usage by pod for sidecar and ambient pods"
+    >}}
+
+{{< image width="100%"
+    link="memory-by-pod-stacked.png"
+    caption="Stacked memory usage by pod for sidecar and ambient pods"
+    >}}
+
+Looking at total CPU and memory utilization, we have to remember that in sidecar scenarios there are 31 sidecar containers required (one Fortio client and 30 httpbin servers), while in ambient, only three ztunnel containers and three waypoint proxies are required. The stacked CPU and memory by pod graphs are excellent at highlighting just how many additional containers are present between scenarios. Memory usage of the Istio dataplane in the ztunnel-only ambient scenario uses *1%* of what is used in sidecar scenarios, and still only 10% when waypoints are added. Looking at CPU, ztunnel once again uses *1%* of what sidecar scenarios require, and 15% when waypoints are deployed.
+
+Finally, let’s consider allocation since the graphs above have only covered usage. Every sidecar resource has a default request of 100 millicores vCPU and 128 MB memory, as well as limits set for 2 vCPUs and 1 GB memory. For simplicity, we assume ztunnels and waypoint proxies have similar requests and limits as their sidecar counterparts – even though every measurement so far has suggested ztunnels will require significantly less. Breaking it down, that’s a 90% reduction in allocated resources with ztunnel, and 80% when waypoints are included.
```

Comment:
A bit worried this simplification ends up meaning that we are just strictly comparing "number of proxies". Which is a bit of a simplistic model. I don't feel great about saying 1 ztunnel == 1 sidecar == 1 waypoint when we know its not true.

Reply:
> 1 ztunnel == 1 sidecar == 1 waypoint 

is your issue with the assumption that waypoints and ztunnels are using the same request/limits as sidecars even though performance data so far suggests otherwise?  That I should be using different values for ztunnel in particular?

Reply:
Yes. Otherwise this is just counting proxies, which is a much simpler blog post and could be 2-3 sentences long